### PR TITLE
add CenterPoint.

### PR DESCRIPTION
**Pull request by tanzhenyu**

### CODE REVIEW

File: keras_cv/layers/object_detection3d/heatmap_decoder.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        loc = tf.stack(delta, axis=-1)
+        box = tf.concat([loc, lwh, heading[:, tf.newaxis]], axis=-1)
+        return box
+
+
+class HeatmapDecoder(tf.keras.layers.Layer):
+    """A Keras layer that decodes predictions of an 3d object detection model.
+
+    Arguments:
+      max_pool_size:
```
Comment (lengzq) [inline keras_cv/layers/object_detection3d/heatmap_decoder.py:102]: `max_pool_size:XXX`

File: keras_cv/layers/object_detection3d/heatmap_decoder.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        self.num_head_bin = num_head_bin
+        self.anchor_size = anchor_size
+        self.max_pool_size = max_pool_size
+        self.max_num_box = max_num_box
+        self.heatmap_threshold = heatmap_threshold
+        self.voxel_size = voxel_size
+        self.spatial_size = spatial_size
+        self.built = True
+
+    def call(self, prediction):
```
Comment (lengzq) [inline keras_cv/layers/object_detection3d/heatmap_decoder.py:132]: prediction: tf.Tensor

Comment (lengzq) [reply keras_cv/layers/object_detection3d/heatmap_decoder.py:132]: -> Tuple[tf.Tensor, tf.Tensor, tf.Tensor]

File: keras_cv/models/object_detection3d/center_point.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+    def call(self, feature: tf.Tensor, training: bool) -> List[tf.Tensor]:
+        del training
+        outputs = {}
+        for head_name in self._head_names:
+            outputs[head_name] = self._heads[head_name](feature)
+        return outputs
+
+
+class MultiHeadCenterPoint(tf.keras.Model):
```
Comment (lengzq) [inline keras_cv/models/object_detection3d/center_point.py:82]: MultiHeadCenterPillars

**Comment by lengzq** [comment]
Thanks! Please change center_point.py -> center_pillars.py because the implementation is not the same

File: keras_cv/layers/object_detection3d/heatmap_decoder.py (hunk 1 above)
Comment (tanzhenyu) [reply keras_cv/layers/object_detection3d/heatmap_decoder.py]: Done

File: keras_cv/layers/object_detection3d/heatmap_decoder.py (hunk 2 above)
Comment (tanzhenyu) [reply keras_cv/layers/object_detection3d/heatmap_decoder.py]: Done

File: keras_cv/models/object_detection3d/center_point.py (hunk 3 above)
Comment (tanzhenyu) [reply keras_cv/models/object_detection3d/center_point.py]: Done

File: keras_cv/layers/object_detection3d/heatmap_decoder.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    """Decode bin heading.
+
+    Computes the box heading (orientation) by decoding the bin predictions. The
+    predictions should contain bin classification scores (first num_bin scores)
+    and corresponding bin residuals (the following num_bin scores).
+
+    Args:
+      predictions: Prediction scores tensor with size [N, num_bin*2] predictions =
+        [:, bin_1, bin_2, ..., bin_k, res_1, res_2, ..., res_k], where k is the
+        number of bins and N is the number of boxes
```
Comment (lengzq) [inline keras_cv/layers/object_detection3d/heatmap_decoder.py:34]: Missing period (.). Many docstrings in this PR miss period.

File: keras_cv/models/object_detection3d/center_pillar_test.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+from keras_cv.models.__internal__.unet import UNet
+from keras_cv.models.object_detection3d.center_pillar import MultiClassDetectionHead
+from keras_cv.models.object_detection3d.center_pillar import MultiClassHeatmapDecoder
+from keras_cv.models.object_detection3d.center_pillar import MultiHeadCenterPillar
+
+down_block_configs = [(128, 6), (256, 2), (512, 2)]
+up_block_configs = [512, 256, 256]
+
+
+class CenterPillarsTest(tf.test.TestCase):
```
Comment (lengzq) [inline keras_cv/models/object_detection3d/center_pillar_test.py:28]: `CenterPillarTest` no `s`

**Review by lengzq — CHANGES_REQUESTED** [review]
Thanks! Overall looks great!!

File: keras_cv/layers/object_detection3d/heatmap_decoder.py (hunk 4 above)
Comment (tanzhenyu) [reply keras_cv/layers/object_detection3d/heatmap_decoder.py:34]: Added

File: keras_cv/models/object_detection3d/center_pillar_test.py (hunk 5 above)
Comment (tanzhenyu) [reply keras_cv/models/object_detection3d/center_pillar_test.py:28]: Done

**Review by lengzq — APPROVED** [review]

**Pull request merged**